crates/revm/src/executor.rs

```diff
@@ -325,6 +325,7 @@ where
 pub fn increment_account_balance<DB>(
     db: &mut CacheDB<DB>,
     post_state: &mut PostState,
+    block_number: BlockNumber,
```

Comment:
I see why we need this now, because the posstate no longer tracks the `current_transition_id`.

I think this is fine, using the block number as an argument makes it less susceptible to bugs